### PR TITLE
Add a severity to messages

### PR DESCRIPTION
**Pull request by jeremycline**
Each message can now have a severity associated with it, both at a class
level and on a message-by-message basis.

Signed-off-by: Jeremy Cline <jcline@redhat.com>

### CODE REVIEW

**Comment by jeremycline** [comment]
> While I believe that the specific values of severities might be hard to determine, this code looks reasonable to me.

Do you mean for a particular message it's difficult to say what the severity is? I agree and I expect we'll have plenty of spirited discussions stemming from someone getting notifications they're not interested it. It's not perfect, but hopefully that will be a guiding force for the "right" severity level.

@abompard, I think I'm going to go ahead and merge this for a beta release (which includes all the fixed-up auth stuff), but when you get back from PTO, please review this and we can adjust it before a stable release is tagged.

File: fedora_messaging/message.py [hunk 1]
```diff
@@ -288,18 +340,18 @@ def validate(self):
...
+                'Validating message headers "%r" with schema "%r"',
+                self._headers,
+                schema,
+            )
+            jsonschema.validate(self._headers, schema)
+        for schema in (self.body_schema, Message.body_schema):
+            _log.debug(
+                'Validating message body "%r" with schema "%r"', self._body, schema
+            )
+            jsonschema.validate(self._body, schema)
```
Comment (abompard) [inline fedora_messaging/message.py]: You're doing this so people don't override the base schemas too much? Are you thinking of a specific use case?

File: fedora_messaging/tests/unit/test_message.py [hunk 2]
```diff
@@ -22,11 +22,44 @@
...
+        self.assertRaises(
+            exceptions.ValidationError, message.get_message, "", msg._properties, b"{}"
+        )
+
+    def test_missing_headers(self):
+        """Assert missing headers results in a default message."""
+        msg = message.Message()
+        msg._headers = None
+        expected_message = message.Message()
+        expected_message.id = msg.id
```
Comment (abompard) [inline fedora_messaging/tests/unit/test_message.py]: You don't seem to be using this variable.

File: fedora_messaging/message.py (hunk 1 above)
Comment (jeremycline) [reply fedora_messaging/message.py]: I was thinking that for the headers, it'd be good to validate we added the expected schema, timestamp, etc. keys and the easiest way would be to add it to the schema, but then I wanted people to have the ability to expand upon it. With the body I think it's reasonable to demand all messages be an object, so this makes it enforced.

I can't really think of a use-case for users to actually add headers at the moment so maybe that should all be an internal thing, though.

File: fedora_messaging/tests/unit/test_message.py (hunk 2 above)
Comment (jeremycline) [reply fedora_messaging/tests/unit/test_message.py]: Ah, good eye

**Comment by jeremycline** [comment]
> This looks good! Should the tutorial be updated to ask people to set the default severity?

I'll do that

**Pull request merged**